File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s.isEmpty()) {
            return true;
        }
        
        int start = 0, end = s.length()-1;
        char cStart, cEnd; 
        while (start <= end) {
            cStart = Character.toLowerCase(s.charAt(start));
            cEnd = Character.toLowerCase(s.charAt(end));
           
            if (!Character.isLetterOrDigit(cStart)) {
                start++;
            } else if (!Character.isLetterOrDigit(cEnd)) {
                end--;
            } else {
                if (cStart != cEnd) {
                    return false;
                }
                start++;
                end--;
            }
            
        }
        
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        cs = ""
        
        for char in s:
            if char.isalnum():
                if char.isnumeric():
                    cs += char
                else:
                    cs += char.lower()
        print(cs)
        start = 0
        end = len(cs) - 1
        mid = int(len(cs)/2)
        if cs:
            for i in range(start, mid+1):
                if cs[i] != cs[end]:
                    return False

                end -= 1
        return True